lang/java/reef-utils/src/test/java/org/apache/reef/util/MultiAsyncToSyncTest.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Performs an asynchronous increment of an Integer.
+ */
+final class AsynchronousIncrementer implements Callable<Integer> {
+  private static final Logger LOG = Logger.getLogger(MultiAsyncToSync.class.getName());
+  private final int sleepTimeMillis;
+  private final int input;
+  private final long identifier;
+  private final MultiAsyncToSync blocker;
+
+  /**
+   * Instantiate an incrementer with specific job parameters.
+   * @param input The input parmeter for the work.
+   * @param identifier The identifier of the caller to wake on completion.
+   * @param sleepTimeMillis How long to work.
+   * @param blocker The MultiAsyncToSync object which is holding the blocked client.
+   */
+  AsynchronousIncrementer(final Integer input, final long identifier,
```

Comment:
Because the generic parameter of Java's FutureTask class has to be a subclass of object.